### PR TITLE
Changes to guarantee that --version works always

### LINKED ISSUE #169: The --version option doesn't work when using the binary
@marinaglancy recently detected that the binary version of the tool doesn't show correct information when using the `--version`option.
```
$ bin/moodle-plugin-ci --version
Moodle Plugin CI @package_version@
```

Curiously, it seems to work when used with the phar package:
```
$ php build/moodle-plugin-ci.phar --version
Moodle Plugin CI 3.2.6-4-gb88bd4c
```

This is about to understand why that's happening and fix it, by getting consistent and correct results.

Ciao :-)

### PR DESCRIPTION
**Pull request by stronk7**
IMPORTANT: This needs to be merged after #171 is merged and this has been rebased (composer regenerated).

With this change:

1. moodle-plugin-ci.phar will continue working the same,
   with `--version` always returning the correct (git describe)
   version used to build the phar.
2. bin/moodle-plugin-ci will also show the correct (git describe)
   version, when it's run from a git directory. If not git is detected
   the version specified in `$version` will be used.

Plus:

- In order to ensure that the `$version` is updated on every release a
  new check has been added to `make validate`. It will compare the
  latest version documented in the 'docs/CHANGELOG.md' file with the
  version in the `bin/moodle-plugin-ci` file.
- Release docs have been amended to remember to update the version
  in `bin/moodle-plugin-ci` when making a release.

Fixes #169

### CODE REVIEW

File: bin/moodle-plugin-ci
Comment (kabalin) [reply bin/moodle-plugin-ci:49]: Thinking more, may be version needed to be defined in `composer.json` instead of php script, which can be parsed for version displaying purposes here? That might have implications though (e.g. on composer versioning).

Comment (stronk7) [reply bin/moodle-plugin-ci:49]: About to make it a constant on top of file, I'm ok with that, good idea.

About to add it to composer... I just followed what other packages out there are doing (phpunit...) and read some comments about keeping them apart from composer explicitly.

**Comment by stronk7** [comment]
I've added a fixup commit to the branch, performing the move from straight-harcoded in the code to new `MOODLE_PLUGIN_CI_VERSION` constant/define.

Will squash and amend commit message details if this gets its thumbs up (remember it's blocked by https://github.com/moodlehq/moodle-plugin-ci/pull/171).

Ciao :-)

**Review by kabalin — APPROVED** [review]

**Comment by kabalin** [comment]
Looks good @stronk7, thanks!

**Comment by stronk7** [comment]
Thanks @kabalin. I've already squashed the commits. Once #171 is landed, I'll rebase again, rebuilding composer lock file and merge this.

**Comment by stronk7** [comment]
I've rebased this on top of the just merged #171. Note it comes with an update of PHP_CodeSniffer to 3.7.1 that was released 10 days ago. Other than that, not changes, so far.

**Comment by stronk7** [comment]
I see above that, somehow, I "dismissed [kabalin](https://github.com/kabalin)’s stale review", and I don't know what that means.

I assume it already got your thumbs up and, as far as all the testing is passing... I'm going to self-merge this now.

Aiming to a new release today/tomorrow.

Ciao :-)

**Comment by jrchamp** [comment]
Explanation: When you pushed new code, the review was "stale"/old. GitHub has a setting that automatically dismisses reviews if the code does not match the version that was reviewed. For repositories that require at least one approving review, this prevents unreviewed code from being merged. As you are one of the administrators and reviews are optional on this repository, it's not necessary.

Hope that helps! ❤️

**Comment by stronk7** [comment]
Thanks, @jrchamp !

**Pull request merged**
